pkg/codegen/convert/plugin_mapper.go

```diff
@@ -173,12 +173,96 @@ func NewPluginMapper(ws Workspace,
 	}, nil
 }
 
+// getMappingForPlugin calls GetMapping on the given plugin and returns it's result. This currently takes the
+// target provider name as an argument so that if we don't find the mapping for "terraform" we can try again
+// for "tf".
+func (l *pluginMapper) getMappingForPlugin(targetProvider string, pluginSpec mapperPluginSpec) ([]byte, string, error) {
+	providerPlugin, err := l.providerFactory(pluginSpec.name, &pluginSpec.version)
+	if err != nil {
+		// We should maybe be lenient here and ignore errors but for now assume it's better to fail out on
+		// things like providers failing to start.
+		return nil, "", fmt.Errorf("could not create provider '%s': %w", pluginSpec.name, err)
+	}
+	contract.IgnoreClose(providerPlugin)
+
+	data, mappedProvider, err := providerPlugin.GetMapping(l.conversionKey)
+	if err != nil {
+		// This was an error calling GetMapping, not just that GetMapping returned a nil result. It's fine for
+		// GetMapping to return (nil, nil) as that simply indicates that the plugin doesn't have a mapping for
+		// the requested key.
+		return nil, "", fmt.Errorf("could not get mapping for provider '%s': %w", pluginSpec.name, err)
+	}
+	// A provider returns empty if it didn't have a mapping
+	if mappedProvider != "" && len(data) != 0 {
+		// If this was the provider we we're looking for we can now return it
+		if mappedProvider == targetProvider {
+			return data, mappedProvider, nil
+		}
+	}
+	// TODO: Temporary hack to work around the fact that most of the plugins return a mapping for "tf" but
+	// not "terraform" but they're the same thing.
+	if l.conversionKey == "terraform" {
+		// Copy-pasta of the above _but_ we'll delete this whole if block once the plugins have had a
+		// chance to update.
```

Comment:
Not actionable, but this comment existed in the old version too.
Just out of curiosity, is this really a temporary hack TODO or is it more permanent now?
Because it's trivial to drop the duplication:

```go
conversionKeys := []string{l.conversionKey}
if l.conversionKey == "terraform" { conversionKeys = append(conversionKeys, "tf") }

for _, conversionKey := range conversionKeys {
  data, mappedProvider, err := ...
  // ...
}
```

But again: not actionable in this PR either way

Reply:
It's a very temporary hack, should get removed soon. Was just to give providers a chance to update.